labs/bdls.md

```diff
@@ -0,0 +1,30 @@
+---
+layout: default
+title: BDLS
+[//]: # (parent: Labs)
+---
+# Lab Name
+BDLS
+
+# Short Description
+BDLS is an innovative BFT consensus algorithm that features safety and liveness by presenting a mathematically proven secure BFT protocol that is resilient in open networks such as the Internet. With BDLS, we invented a new random beacons to ensure verifiable unpredictability and fairness of validators. More importantly, BDLS overcomes many problems, such as DoS attacks, as well as the deadlock problem caused by unreliable p2p/broadcast channels. These problems are all very relevant to existing realistic open network scenarios, and are the focus of extensive work in improving Internet security, but it is an area largely ignored by most in mainstream BFT protocol design.
+(Paper: https://eprint.iacr.org/2019/1460.pdf)
+
+For this library, to make the runtime behavior of consensus algorithm predictable as function: y = f(x, t), where 'x' is the message it received, and 't' is the time while being called, then'y' is the deterministic status of consensus after 'x' and 't' applied to 'f', it has been designed in a deterministic scheme, without parallel computing, networking, and the correctness of program implementation can be proven with proper test cases.
+
+# Scope of Lab
+BDLS library is planned to be integrated into Hyperledger Fabric as its ordering service, eventually replacing Fabric's Raft ordering service. Other distributed ledger projects under Hyperledger can also benefit from and use BDLS.
+
+# Initial Committers
+
+- https://github.com/yonggewang
+- https://github.com/ahmed82
+
+
+# Sponsor
+Kamlesh Nagware, Chief Technology Officer at Snapper Future Tech and Co-Lead Hyperledger India chapter
+
+# Pre-existing repository
+
+- https://github.com/yonggewang/bdls (main repository)
```

Comment:
Two things:

First,  it looks like none of the commits in this repository contain DCO signoff. Hyperledger requires that all contributed code has a DCO sign off. Looking at these repositories, it appears that this is not the case. There are two options that we normally consider for fixing this:

1. Amend existing commits in the repository to include DCO sign off
2. Manually transfer the code from the existing repo to the newly created repo within the hyperledger-labs GitHub organization.

Second, the license specified in the repo is [BSD-3-Clause License](https://github.com/yonggewang/bdls/blob/master/LICENSE). According to [Hyperledger's Charter](https://www.hyperledger.org/about/charter), this is not an acceptable license.
> Intellectual Property Policy
>
> a. Members agree that all new inbound code contributions to the Hyperledger Foundation shall be made under the Apache License, Version 2.0 (available at http://www.apache.org/licenses/LICENSE-2.0). All contributions shall be accompanied by a Developer Certificate of Origin sign-off (http://developercertificate.org/) that is submitted through a Governing Board and LF-approved contribution process. Such contribution process will include steps to also bind non-Member Contributors and, if not self-employed, their employer, to the licenses expressly granted in the Apache License, Version 2.0 with respect to such contribution.
> 
> b. All outbound code will be made available under the Apache License, Version 2.0.
> 
> c. All documentation will be contributed to and made available by the Hyperledger Foundation under the Creative Commons Attribution 4.0 International License (available at http://creativecommons.org/licenses/by/4.0/).
> 
> d. If an alternative inbound or outbound license is required for compliance with the license for a leveraged open source project or is otherwise required to achieve the Hyperledger Foundation’s mission, the Governing Board may approve the use of an alternative license for specific inbound or outbound contributions on an exception basis. Any exceptions must be approved by a two-thirds vote of the entire Governing Board and the LF and must be limited in scope to what is required for such purpose. Please email [legal@hyperledger.org](mailto:legal@hyperledger.org) to obtain exception approval.
> 
> e. Subject to available Foundation funds, the Hyperledger Foundation may engage The Linux Foundation to determine the availability of, and register, trademarks, service marks, and certification marks, which shall be owned by the LF.

Reply:
1- The license have been update by the repository owner to the Apache License, Version 2.0 as the suggested.
2- The sign-off added by Yongge Wang after update the repository license, please check and let me know if I miss anything.
Thanks for reviewing the PR
Signed-off-by: Ahmed Al Salih a.alsalih2@gmail.com

Reply:
This is an interesting lab project,  this could be added to the Fabric roadmap after successful implementation. Strong support to have more consensus options.